experiments/integrated/global/global_soil_canopy.jl

```diff
@@ -332,7 +332,9 @@ Y.soil.ρe_int .=
         soil_params.earth_param_set,
     )
 Y.soilco2.C .= FT(0.000412) # set to atmospheric co2, mol co2 per mol air
-Y.canopy.hydraulics.ϑ_l.:1 .= plant_ν
+# On the Caltech cluster, this is not permitted, so we allowscalar.
+# This operation is fine on clima cluster
+@allowscalar Y.canopy.hydraulics.ϑ_l.:1 .= plant_ν
```

Comment:
What's the error on the caltech cluster?

Reply:
```
julia --color=yes --project=.buildkite experiments/integrated/global/global_soil_canopy.jl
ERROR: LoadError: Scalar indexing is disallowed.
Invocation of setindex! resulted in scalar indexing of a GPU array.
This is typically caused by calling an iterating implementation of a method.
Such implementations *do not* execute on the GPU, but very slowly on the CPU,
and therefore should be avoided.
If you want to allow scalar iteration, use `allowscalar` or `@allowscalar`
to enable scalar iteration globally or for the operations in question.
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:35
  [2] errorscalar(op::String)
    @ GPUArraysCore /central/scratch/esm/slurm-buildkite/climaland-ci/depot/default/packages/GPUArraysCore/GMsgk/src/GPUArraysCore.jl:155
  [3] _assertscalar(op::String, behavior::GPUArraysCore.ScalarIndexing)
    @ GPUArraysCore /central/scratch/esm/slurm-buildkite/climaland-ci/depot/default/packages/GPUArraysCore/GMsgk/src/GPUArraysCore.jl:128
  [4] assertscalar(op::String)
    @ GPUArraysCore /central/scratch/esm/slurm-buildkite/climaland-ci/depot/default/packages/GPUArraysCore/GMsgk/src/GPUArraysCore.jl:116
  [5] setindex!(A::CUDA.CuArray{Float64, 4, CUDA.DeviceMemory}, v::Float64, I::Int64)
    @ GPUArrays /central/scratch/esm/slurm-buildkite/climaland-ci/depot/default/packages/GPUArrays/OqrUV/src/host/indexing.jl:56
  [6] setindex!
    @ ./subarray.jl:343 [inlined]
  [7] _setindex!
    @ ./abstractarray.jl:1419 [inlined]
  [8] setindex!
    @ ./abstractarray.jl:1396 [inlined]
  [9] set_struct!
    @ /central/scratch/esm/slurm-buildkite/climaland-ci/depot/default/packages/ClimaCore/Bf3Rx/src/DataLayouts/struct.jl:257 [inlined]
 [10] setindex!
    @ /central/scratch/esm/slurm-buildkite/climaland-ci/depot/default/packages/ClimaCore/Bf3Rx/src/DataLayouts/DataLayouts.jl:784 [inlined]
 [11] fill!(data::ClimaCore.DataLayouts.IJF{Float64, 2, SubArray{Float64, 3, CUDA.CuArray{Float64, 4, CUDA.DeviceMemory}, Tuple{Base.Slice{Base.OneTo{Int64}}, Base.Slice{Base.OneTo{Int64}}, UnitRange{Int64}, Int64}, true}}, val::Float64)
    @ ClimaCore.DataLayouts /central/scratch/esm/slurm-buildkite/climaland-ci/depot/default/packages/ClimaCore/Bf3Rx/src/DataLayouts/DataLayouts.jl:734
 [12] fill!(data::ClimaCore.DataLayouts.IJFH{Float64, 2, SubArray{Float64, 4, CUDA.CuArray{Float64, 4, CUDA.DeviceMemory}, Tuple{Base.Slice{Base.OneTo{Int64}}, Base.Slice{Base.OneTo{Int64}}, UnitRange{Int64}, Base.Slice{Base.OneTo{Int64}}}, false}}, val::Float64)
    @ ClimaCore.DataLayouts /central/scratch/esm/slurm-buildkite/climaland-ci/depot/default/packages/ClimaCore/Bf3Rx/src/DataLayouts/DataLayouts.jl:316
 [13] copyto!
    @ /central/scratch/esm/slurm-buildkite/climaland-ci/depot/default/packages/ClimaCore/Bf3Rx/src/DataLayouts/broadcast.jl:479 [inlined]
 [14] copyto!
    @ /central/scratch/esm/slurm-buildkite/climaland-ci/depot/default/packages/ClimaCore/Bf3Rx/src/Fields/broadcast.jl:408 [inlined]
 [15] materialize!
    @ ./broadcast.jl:914 [inlined]
 [16] materialize!(dest::ClimaCore.Fields.Field{ClimaCore.DataLayouts.IJFH{Float64, 2, SubArray{Float64, 4, CUDA.CuArray{Float64, 4, CUDA.DeviceMemory}, Tuple{Base.Slice{Base.OneTo{Int64}}, Base.Slice{Base.OneTo{Int64}}, UnitRange{Int64}, Base.Slice{Base.OneTo{Int64}}}, false}}, ClimaCore.Spaces.SpectralElementSpace2D{ClimaCore.Grids.SpectralElementGrid2D{ClimaCore.Topologies.Topology2D{ClimaComms.SingletonCommsContext{ClimaComms.CUDADevice}, ClimaCore.Meshes.EquiangularCubedSphere{ClimaCore.Domains.SphereDomain{Float64}, ClimaCore.Meshes.NormalizedBilinearMap}, CartesianIndices{3, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}, Base.OneTo{Int64}}}, LinearIndices{3, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}, Base.OneTo{Int64}}}, CUDA.CuArray{Tuple{Int64, Int64, Int64, Int64, Bool}, 1, CUDA.DeviceMemory}, Vector{Tuple{Int64, Int64, Int64, Int64, Bool}}, CUDA.CuArray{Tuple{Int64, Int64}, 1, CUDA.DeviceMemory}, CUDA.CuArray{Int64, 1, CUDA.DeviceMemory}, CUDA.CuArray{Tuple{Bool, Int64, Int64}, 1, CUDA.DeviceMemory}, CUDA.CuArray{Int64, 1, CUDA.DeviceMemory}, CUDA.CuArray{Int64, 1, CUDA.DeviceMemory}, @NamedTuple{}, CUDA.CuArray{Tuple{Int64, Int64}, 1, CUDA.DeviceMemory}}, ClimaCore.Quadratures.GLL{2}, ClimaCore.Geometry.SphericalGlobalGeometry{Float64}, ClimaCore.DataLayouts.IJFH{ClimaCore.Geometry.LocalGeometry{(1, 2), ClimaCore.Geometry.LatLongPoint{Float64}, Float64, StaticArraysCore.SMatrix{2, 2, Float64, 4}}, 2, CUDA.CuArray{Float64, 4, CUDA.DeviceMemory}}, ClimaCore.DataLayouts.IJFH{Float64, 2, CUDA.CuArray{Float64, 4, CUDA.DeviceMemory}}, ClimaCore.DataLayouts.IFH{ClimaCore.Geometry.SurfaceGeometry{Float64, ClimaCore.Geometry.UVVector{Float64}}, 2, CUDA.CuArray{Float64, 3, CUDA.DeviceMemory}}, @NamedTuple{}}}}, bc::Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Nothing, typeof(identity), Tuple{Float64}})
    @ Base.Broadcast ./broadcast.jl:911
 [17] top-level scope
    @ /central/scratch/esm/slurm-buildkite/climaland-ci/4335/climaland-ci/experiments/integrated/global/global_soil_canopy.jl:335
in expression starting at /central/scratch/esm/slurm-buildkite/climaland-ci/4335/climaland-ci/experiments/integrated/global/global_soil_canopy.jl:335
🚨 Error: The command exited with status 1
```

the same exact line runs on clima

Reply:
Weird, might be a depot thing. I don't think they should behave differently

Reply:
Im not sure. Im just going to remove the GPU run on caltech I think.

Reply:
Can we open an issue and just copy the error and where it comes from? Even if we don't address/prioritize a fix, it would be nice to have this documented

Reply:
I'll do this!